### ISSUE TITLE
[bug] Unable to compute sha256 via @aws-crypto/sha256-browser slicing when file is larger than 2g

### ISSUE THREAD

**Issue by Woodii1998** [aws/aws-sdk-js-crypto-helpers#769]
Error when calculating sha256 for files larger than 2g

![img_v2_29f632a3-7df4-40dc-ae3c-1a6d68fc523g](https://github.com/aws/aws-sdk-js-crypto-helpers/assets/33593572/3f5a5307-9648-400c-9c0f-9912eb651093)

example：https://codesandbox.io/s/mystifying-thunder-53gsf3?file=/src/App.js

**Comment by lucasmcdonald3** [comment]
Hi @Woodii1998,

Thanks for bringing this to our attention.

You are correct that you will run into issues hashing files of this size with this library.

The underlying limitation is that JavaScript's ArrayBuffer type nominally supports array sizes up to `2^31-1`.
(You may encounter smaller limits depending on your environment's limitations.)
The library we use for hashing uses ArrayBuffers to store data.
If your updates result in an array that exceeds the supported ArrayBuffer size,
you will be unable to hash the file. 
I'd also note that the library's [digest](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest) method requires that the entire digest input is stored in-memory (i.e. no streaming/batching the digest call).

In summary: we cannot suggest an easy workaround that will produce a valid SHA256 hash for files that exceed the maximum ArrayBuffer size.

Your approach to handling this limitation depends on your application use case:
* You could limit the size of files you hash to avoid this failure mode entirely
* You could find an alternative library that supports streaming inputs to SHA256

Let us know if you have any questions.

Best,
Lucas
